purchase_blanket_order/models/blanket_orders.py

```diff
@@ -0,0 +1,359 @@
+# Copyright (C) 2018 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero
+
+import odoo.addons.decimal_precision as dp
+
+
+class BlanketOrder(models.Model):
+    _name = 'purchase.blanket.order'
+    _inherit = ['mail.thread']
+    _description = 'Blanket Order'
+
+    @api.model
+    def _default_currency(self):
+        return self.env.user.company_id.currency_id
+
+    @api.model
+    def _default_company(self):
+        return self.env.user.company_id
+
+    name = fields.Char(
+        default='Draft',
+        readonly=True
+    )
+    partner_id = fields.Many2one(
+        'res.partner', string='Vendor', readonly=True,
+        states={'draft': [('readonly', False)]})
+    lines_ids = fields.One2many(
+        'purchase.blanket.order.line', 'order_id', string='Order lines',
+        copy=True)
+    currency_id = fields.Many2one(
+        'res.currency', related='company_id.currency_id', readonly=True)
+    payment_term_id = fields.Many2one(
+        'account.payment.term', string='Payment Terms', readonly=True,
+        states={'draft': [('readonly', False)]})
+    confirmed = fields.Boolean()
+    state = fields.Selection(selection=[
+        ('draft', 'Draft'),
+        ('open', 'Open'),
+        ('expired', 'Expired'),
+    ], compute='_compute_state', store=True, copy=False)
+    validity_date = fields.Date(
+        readonly=True,
+        states={'draft': [('readonly', False)]})
+    date_order = fields.Datetime(
+        readonly=True,
+        required=True,
+        string='Ordering Date',
+        default=fields.Datetime.now,
+        states={'draft': [('readonly', False)]})
+    note = fields.Text(
+        readonly=True,
+        states={'draft': [('readonly', False)]})
+    user_id = fields.Many2one(
+        'res.users', string='Responsible', readonly=True,
+        states={'draft': [('readonly', False)]})
+    company_id = fields.Many2one(
+        'res.company', string='Company', default=_default_company,
+        readonly=True,
+        states={'draft': [('readonly', False)]})
+    purchase_count = fields.Integer(compute='_compute_purchase_count')
+
+    # Fields use to filter in tree view
+    original_qty = fields.Float(
```

Comment:
@ageficent no need to default, Float alwayse default 0.0, 
Change everywhare

https://odoo-new-api-guide-line.readthedocs.io/en/latest/fields.html#float